### PR TITLE
Feature add coin after completed task

### LINKED ISSUE #82: Update coin balance after completing a task

### PR DESCRIPTION
**Pull request by kome12**
## Description

- Add coin to user after completedTask
- Rename getUserByUserId function to GetUserByFireBaseUserId
- Add CreatedDate and LastUpdate to user model

## Closes issue(s)

Closes #82 

## Screenshots

スクリーンショット 2021-09-05 15 25 56

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [X] I have tested this code

### CODE REVIEW

File: controllers/v1/users/users_controller.go [hunk 1]
```diff
@@ -29,7 +33,7 @@ func CreateUser(createdUserPost users_models.Users) (*mongo.InsertOneResult, err
 	return res, insertErr
 }
 
-func GetUserByUserId(createUserId interface{}) (users_models.Users, error) {
+func GetUserByFireBaseUserId(createUserId interface{}) (users_models.Users, error) {
```
Comment (kome12) [inline controllers/v1/users/users_controller.go:36]: Updated the function name so it's more specific

File: models/v1/users/users_models.go [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+)
 
 type Users struct {
 	ID                primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
 	FireBaseUserId    string             `bson:"fireBaseUserId" json:"fireBaseUserId"`
 	NumCoins          int                `bson:"numCoins" json:"numCoins"`
-	FlowerCollections []string           `bson:"flowerCollections" json:"flowerCollections"`
+	FlowerCollections []primitive.ObjectID           `bson:"flowerCollections" json:"flowerCollections"`
+	CreatedDate time.Time `bson:"createdDate" json:"createdDate"`
+	LastUpdate time.Time `bson:"lastUpdate" json:"lastUpdate"`
```
Comment (kome12) [inline models/v1/users/users_models.go:15]: Added the createdDate and lastUpdate date to the user model

File: models/v1/users/users_models.go [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
 
 type Users struct {
 	ID                primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
 	FireBaseUserId    string             `bson:"fireBaseUserId" json:"fireBaseUserId"`
 	NumCoins          int                `bson:"numCoins" json:"numCoins"`
-	FlowerCollections []string           `bson:"flowerCollections" json:"flowerCollections"`
+	FlowerCollections []primitive.ObjectID           `bson:"flowerCollections" json:"flowerCollections"`
```
Comment (kome12) [inline models/v1/users/users_models.go:13]: Changed the type of the flowerCollections to be ObjectIds

**Review by yukari-n-erb — APPROVED** [review]
LGTM!

**Review by conalli — APPROVED** [review]
lgtm!

**Pull request merged**